binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/AbstractKafkaStreamsBinderProcessor.java

```diff
@@ -455,13 +454,21 @@ protected Serde<?> getValueSerde(String inboundName, KafkaStreamsConsumerPropert
 			// Branching based on event type match.
 			final KStream<?, ?>[] branch = stream.branch((key, value) -> matched.getAndSet(false));
 			// Deserialize if we have a branch from above.
-			final KStream<?, Object> deserializedKStream = branch[0].mapValues(value -> valueSerde.deserializer().deserialize(
-					topicObject.get(), headersObject.get(), ((Bytes) value).get()));
+			final KStream<?, Object> deserializedKStream = !kafkaStreamsConsumerProperties.isUseConfiguredSerdeWhenRoutingEvents() ?
+				branch[0].mapValues(value -> valueSerde.deserializer().deserialize(
+					topicObject.get(), headersObject.get(), ((Bytes) value).get())) : (KStream<?, Object>) branch[0];
```

Comment:
Why do we not have to deserialize w/ the configured value serde in the case where `isUseConfiguredSerdeWhenRoutingEvents=true`? You are just passing the branch as-is in that case?

Reply:
Yes, that is right. If it is `true`, it already got deserialized with that `Serde`.